salt/modules/cassandra_cql.py

```diff
@@ -0,0 +1,585 @@
+    :type  port:           int
+    :return:
+    :rtype:
+
+    .. code-block:: bash
+
+        salt 'minion1' cassandra.create_user username=joe password=secret
+
+        salt 'minion1' cassandra.create_user username=joe password=secret superuser=True
+
+        salt 'minion1' cassandra.create_user username=joe password=secret superuser=True contact_points=minion1
+    '''
+    superuser_cql = 'superuser' if superuser else 'nosuperuser'
+    query = "create user if not exists %s with password '%s' %s;" % (username, password, superuser_cql)
+    log.debug('Attempting to create a new user with username=%s superuser=%s', username, superuser_cql)
+    ret = _query(query, contact_points, port, cql_user, cql_pass)
+    # The create user query doesn't actually return anything if the query succeeds.
+    # So if ret is empty, then the user creation worked. Hence returning True for the function.
+    return True if not ret else ret
+
+
+def list_permissions(username=None, resource=None, resource_type='keyspace', permission=None, contact_points=None,
+                     port=None, cql_user=None, cql_pass=None):
+    '''
+    List permissions.
+
+    :param username:       The name of the user to list permissions for.
+    :type  username:       str
+    :param resource:       The resource (keyspace or table), if None, permissions for all resources are listed.
+    :type  resource:       str
+    :param resource_type:  The resource_type (keyspace or table), defaults to 'keyspace'.
+    :type  resource_type:  str
+    :param permission:     A permission name (e.g. select), if None, all permissions are listed.
+    :type  permission:     str
+    :param contact_points: The Cassandra cluster addresses, can either be a string or a list of IPs.
+    :type  contact_points: str | list[str]
+    :param cql_user:       The Cassandra user if authentication is turned on.
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :return:               Dictionary of permissions.
+    :rtype:                dict
+
+    .. code-block:: bash
+        salt 'minion1' cassandra.list_permissions
+
+        salt 'minion1' cassandra.list_permissions username=joe resource=test_keyspace permission=select
+
+        salt 'minion1' cassandra.list_permissions username=joe resource=test_table resource_type=table \
+        permission=select contact_points=minion1
+    '''
+    cql_template = "list %s on %s"
+
+    keyspace_cql = "%s %s" % (resource_type, resource) if resource else "all keyspaces"
+    permission_cql = "%s permission" % permission if permission else "all permissions"
+
+    query = cql_template % (permission_cql, keyspace_cql)
+    if username:
+        query = "%s of %s" % (query, username)
+    log.debug('Attempting to list permissions with query "%s"', query)
```

Comment:
That's rather noise. I would move it behind the line below (`ret = _query....`) and if `ret` is empty, then would say so in the debug log: query X returns no result.